frontend/src/components/Collections/components/Grid/components/Row/DatasetRow/utils.ts

```diff
@@ -55,3 +89,218 @@ export function checkIfCancelled(datasetStatus: DatasetUploadStatus): boolean {
     datasetStatus.upload_status === UPLOAD_STATUS.CANCELED
   );
 }
+
+export function useCheckCollectionPopulated({
+  invalidateCollectionQuery,
+  isNamePopulated,
+  upload_progress,
+}: {
+  invalidateCollectionQuery: () => void;
+  isNamePopulated: boolean;
+  upload_progress: number;
+}) {
+  useCheckCollection({
+    invalidateCollectionQuery,
+    shouldFetch: upload_progress === 1 && !isNamePopulated,
+  });
+}
+
+type FORMAT_KEYS =
+  | "conversion_loom_status"
+  | "conversion_anndata_status"
+  | "conversion_cxg_status"
+  | "conversion_rds_status";
+
+const CONVERSION_STATUS_FORMAT_KEYS = [
+  "conversion_loom_status",
+  "conversion_anndata_status",
+  "conversion_cxg_status",
+  "conversion_rds_status",
+] as FORMAT_KEYS[];
+
+export function useCheckCollectionFormatsPopulated({
```

Comment:
We only want to fetch collection if we detect there are more formats converted than what's available